packages/stacked_generator/lib/src/generators/router/generator/route_generator_helper.dart

```diff
@@ -0,0 +1,245 @@
+import 'package:stacked_generator/src/generators/base_generator.dart';
+import 'package:stacked_generator/src/generators/router/route_config/route_config.dart';
+import 'package:stacked_generator/utils.dart';
+
+mixin RouteGeneratorHelper on BaseGenerator {
+  void generateImports(List<RouteConfig> routes) {
+    final validImports = routes.map((rc) {
+      return rc.registerImports();
+    }).where((imports) {
+      return imports.isNotEmpty;
+    }).fold<Set<String>>({}, (previousValue, element) {
+      return {...previousValue, ...element};
+    });
+
+    final dartImports =
+        validImports.where((element) => element.startsWith('dart')).toSet();
+    sortAndGenerate(dartImports);
+    newLine();
+
+    var packageImports =
+        validImports.where((element) => element.startsWith('package')).toSet();
+    packageImports.add("package:stacked/stacked.dart");
+    sortAndGenerate(packageImports);
+    newLine();
+
+    var rest = validImports.difference({...dartImports, ...packageImports});
+    sortAndGenerate(rest);
+  }
+
+  void generateRoutesConstantsMap(
+      List<RouteConfig> routes, String routesClassName) {
+    writeLine('class $routesClassName {');
+    var allNames = <String>{};
+    routes.forEach((r) {
+      final routeName = r.name;
+      final path = r.pathName;
+
+      if (path.contains(':')) {
+        // handle template paths
+        writeLine("static const String _$routeName = '$path';");
+        allNames.add('_$routeName');
+        var params = RegExp(r':([^/]+)').allMatches(path).map((m) {
+          var match = m.group(1);
+          if (match!.endsWith('?')) {
+            return "dynamic  ${match.substring(0, match.length - 1)} = ''";
+          } else {
+            return "@required  dynamic $match";
+          }
+        });
+        writeLine(
+          "static String $routeName({${params.join(',')}}) => '${path.replaceAllMapped(RegExp(r'([:])|([?])'), (m) {
+            if (m[1] != null) {
+              return '\$';
+            } else {
+              return '';
+            }
+          })}';",
+        );
+      } else {
+        allNames.add(routeName);
+        writeLine("static const String $routeName = '$path';");
+      }
+    });
+    writeLine("static const all = <String>{");
+    allNames.forEach((name) => stringBuffer.write('$name,'));
+    stringBuffer.write("};");
+    writeLine('}');
+  }
+
+  void generateRouterClass(
+    List<RouteConfig> routes,
+    String routerClassName,
+    String routesClassName,
+  ) {
+    writeLine('\nclass $routerClassName extends RouterBase {');
+
+    writeLine('''
+     @override
+     List<RouteDef> get routes => _routes;
+     final _routes = <RouteDef>[
+     ''');
+    for (var route in routes) {
+      _generateRouteTemplates(route, routesClassName);
+    }
+    writeLine();
+    stringBuffer.write('];');
```

Comment:
Why do some lines write directly to the stringBuffer and others use the write and writeLine function? 

We should only use on style of writing.

Reply:
True I'll fix it